docs/deployments/predictors.md

```diff
@@ -0,0 +1,310 @@
+                ]
+            ]
+        )
+
+        # Run the prediction
+        output = self.model(input_tensor)
+
+        # Translate the model output to the corresponding label string
+        return labels[torch.argmax(output[0])]
+```
+
+## TensorFlow Predictor
+
+### Interface
+
+```python
+class TensorFlowPredictor:
+    def __init__(self, tensorflow_client, config):
+        """Called once before the API becomes available. Performs setup such as downloading/initializing a vocabulary.
+
+        Args:
+            tensorflow_client: TensorFlow client which is used to make predictions. This should be saved for use in predict().
+            config: Dictionary passed from API configuration (if specified).
+        """
+        self.client = tensorflow_client
+        # Additional initialization may be done here
+
+    def predict(self, payload):
+        """Called once per request. Preprocesses the request payload (if necessary), runs inference (e.g. by calling self.client.predict(model_input)), and postprocesses the inference output (if necessary).
+
+        Args:
+            payload: The parsed JSON request payload.
+
+        Returns:
+            Prediction or a batch of predictions.
+        """
+        pass
+```
+
+<!-- CORTEX_VERSION_MINOR -->
+Cortex provides a `tensorflow_client` to your Predictor's constructor. `tensorflow_client` is an instance of [TensorFlowClient](https://github.com/cortexlabs/cortex/tree/master/pkg/workloads/cortex/lib/client/tensorflow.py) that manages a connection to a TensorFlow Serving container to make predictions using your model. It should be saved as an instance variable in your Predictor, and your `predict()` function should call `tensorflow_client.predict()` to make an inference with your exported TensorFlow model. Preprocessing of the JSON payload and postprocessing of predictions can be implemented in your `predict()` function as well.
+
+For proper separation of concerns, it is recommended to use the constructor's `config` paramater for information such as configurable model parameters or download links for initialization files. You define `config` in your [API configuration](api-configuration.md), and it is passed through to your Predictor's constructor.
+
+### Pre-installed packages
+
+The following Python packages are pre-installed in TensorFlow Predictors and can be used in your implementations:
+
+```text
+boto3==1.10.45
+dill==0.3.1.1
+msgpack==0.6.2
+numpy==1.18.0
+requests==2.22.0
+opencv-python==4.1.2.30
+pyyaml==5.3
+tensor2tensor==1.15.4
+tensorflow-hub==0.7.0
+tensorflow==2.1.0
+```
+
+<!-- CORTEX_VERSION_MINOR -->
+The pre-installed system packages are listed in [images/tf-api/Dockerfile](https://github.com/cortexlabs/cortex/tree/master/images/tf-api/Dockerfile).
+
+If your application requires additional dependencies, you can install additional [Python packages](python-packages.md) and [system packages](system-packages.md).
+
+### Examples
```

Comment:
I would place the examples with a sample implementation of the predictor before show pre-installed packages.